### PR TITLE
Fixing arithmetic ring issue

### PR DESCRIPTION
**Pull request by jlevallois**
# PR Description

The aim of this PR is to remove arithmetic ring issue when we change the integer type of `SpaceND`/`KhalimskySpaceND`.
By convention, we use `SpaceND<dim, int32_t>` and `KhalimskySpaceND<dim, int32_t>`.
But, in some case (if we want to reduce the memory footprint for example), we could be led to use `int16_t` instead of `int32_t` (or other).
The issue is that some algorithms are hard-coded to use `int32_t`. This PR try to resolve this issue.

*(I'm sorry, the reviewing process will not be easy because my editor has deleted some unnecessary space in comments ...)*

## Critical files modified

These files need to be closely reviewed.

- `AlphaThickSegmentComputer.h`
- `PowerMap.h` / `PowerMap.ih`
- `VoronoiMap.h` / `VoronoiMap.ih`
- `RigidTransformation3D.h`
- `Display2DFactory.ih`
- `DrawWithDisplay3DModifier.h`
- `BasicPointFunctors.h`

## Test files modified

- `testAlphaThickSegmentComputer.cpp`
- `testFrechetShortcut.cpp`
- `testFreemanChain.cpp`
- `testDistanceTransformationND.cpp`
- `testImageSimple.cpp`
- `testImageSpanIterators.cpp`
- `testSliceImageFromFunctor.cpp`
- `testDigitalSet.cpp`
- `testHyperRectDomain-snippet.cpp`
- `testHyperRectDomain.cpp`
- `testCellularGridSpaceND.cpp`
- `testSurfaceHelper.cpp`

# Checklist

- [N/A] Unit-test of your feature with [Catch](http://dgtal.org/doc/stable/moduleCatch.html).
- [N/A] Doxygen documentation of the code completed (classes, methods, types, members...)
- [N/A] Documentation module page added or updated.
- [x] New entry in the [ChangeLog.md](https://github.com/DGtal-team/DGtal/blob/master/ChangeLog.md) added.
- [x] No warning raised in Debug ```cmake``` mode (otherwise, Travis C.I. will fail).
- [x] All continuous integration tests pass (Travis & appveyor)

:octocat:

### CODE REVIEW

File: src/DGtal/geometry/curves/AlphaThickSegmentComputer.h [hunk 1]
```diff
@@ -130,89 +130,89 @@ class AlphaThickSegmentComputer
...
   typedef typename InputPointContainer::size_type Size;
   typedef typename InputPointContainer::const_iterator ContainerConstIterator;
   typedef typename DGtal::functions::Hull2D::ThicknessDefinition ThicknessDef;
-  
+
   typedef typename InputPointContainer::iterator Iterator;
   typedef TConstIterator ConstIterator;
-  typedef ParallelStrip< SpaceND< 2,  DGtal::int32_t > ,true,true> Primitive;  
-  
+  typedef ParallelStrip< SpaceND< 2 > ,true,true> Primitive;
```
Comment (dcoeurjo) [inline src/DGtal/geometry/curves/AlphaThickSegmentComputer.h]: why did you change this one ? 
Even if int32_t is already the default type for SpacND, maybe the author as a good reason to want a int32_t type (if for instance we change the default type in ```SpaceND```

Comment (jlevallois) [inline src/DGtal/geometry/curves/AlphaThickSegmentComputer.h]: Agree, in this case I need to change `testAlphaThickSgementComputer` who uses `Z2i::Point`/`Z2i::Domain`

Comment (dcoeurjo) [inline src/DGtal/geometry/curves/AlphaThickSegmentComputer.h]: No I think it should be ````SpaceND<2, InputPoint::Coordinate>``` here

**Comment by dcoeurjo** [comment]
nice PR BTW
gif-keyboard-11411730673171660395

File: src/DGtal/geometry/curves/AlphaThickSegmentComputer.h (hunk 1 above)
Comment (jlevallois) [inline src/DGtal/geometry/curves/AlphaThickSegmentComputer.h]: We can't, because of `AlphaThickSegmentComputer< Z2i::RealPoint>` in `testAlphaThickSgementComputer` (line 199)

Comment (dcoeurjo) [inline src/DGtal/geometry/curves/AlphaThickSegmentComputer.h]: Ok I'll have a look.

Comment (jlevallois) [inline src/DGtal/geometry/curves/AlphaThickSegmentComputer.h]: It works when hardcoding `ParallelStrip<SpaceND< 2, int32_t>, ...>` because his `operator()` has a surcharge for `Point` and `RealPoint` (always double) from `SpaceND`.

Comment (dcoeurjo) [inline src/DGtal/geometry/curves/AlphaThickSegmentComputer.h]: Ok... And there is no implicit cast double -> int when used ?

Comment (jlevallois) [inline src/DGtal/geometry/curves/AlphaThickSegmentComputer.h]: `double` doesn't follow the concept `CInteger`, require by `SpaceND`

Comment (dcoeurjo) [inline src/DGtal/geometry/curves/AlphaThickSegmentComputer.h]: Yeah of course. I meant in the code, are input points casted to "primitive" points at some places (which would mean an implicit cast if input points are RealPoints)

**Comment by dcoeurjo** [comment]
thanks again @jlevallois 
Can you please revert your edit in Alpha...  (reset the int32_t template type) and  add an entry to the changeling ?
Beside, the PR is fine, thanks again

**Comment by jlevallois** [comment]
It's done :)

**Comment by dcoeurjo** [comment]
Thanks Merging
